src/Agent.Worker/Release/Artifacts/GitHubArtifact.cs

```diff
@@ -52,6 +52,9 @@ public class GitHubArtifact : AgentService, IArtifactExtension
             gitHubEndpoint.Data.Add(Constants.EndpointData.SourcesDirectory, localFolderPath);
             gitHubEndpoint.Data.Add(Constants.EndpointData.SourceBranch, gitHubDetails.Branch);
             gitHubEndpoint.Data.Add(Constants.EndpointData.SourceVersion, artifactDefinition.Version);
+            gitHubEndpoint.Data.Add(WellKnownEndpointData.CheckoutSubmodules, gitHubDetails.CheckoutSubmodules);
+            gitHubEndpoint.Data.Add("fetchDepth", gitHubDetails.FetchDepth);
```

Comment:
I suppose adding them to WellKnownEndpointData will require change in MSENG & a client update to refer it from here?

Can you do this change in MSENG and later on reference the WellKnownEndpointData constant instead of using the hard-coded constant?